# **Week 6: Discrete distributions**

```
.------------------------------------.
|   __  ____  ______  _  ___ _____   |
|  |  \/  \ \/ / __ )/ |/ _ \___  |  |
|  | |\/| |\  /|  _ \| | | | | / /   |
|  | |  | |/  \| |_) | | |_| |/ /    |
|  |_|  |_/_/\_\____/|_|\___//_/     |
'------------------------------------'

```

Today's workshop is on discrete distributions. We will have a combination of pen-and-paper questions and learn how to simulate and do calculations with discrete random variables in R.

## Warm-up activity
In each of the following situations, suggest a distribution that could be suitable for the random variable.

a) The number of trolleys with defective wheels at a supermarket

b) The number of shoppers arriving in 5 minutes

c) The number of rainy days in April.

d) The number of planes landing at Brisbane airport until you see two A380's.

e) The number of times you roll a die before rolling a 6.

f) The number of potholes in 10 km of freeway north of Brisbane.

g) Whether or not the lift in Y block is working.

h) The suit of a card drawn from a standard deck of 52 cards.

<details>
<summary>▶️ Click to show the solution</summary>

a) Binomial

b) Poisson

c) Binomial

d) Negative Binomial

e) Geometric

f) Poisson

g) Bernoulli

h) Discrete Uniform

</details>

## **Pre-Configurating the Notebook**

### **Switching to the R Kernel on Colab**

By default, Google Colab uses Python as its programming language. To use R instead, you’ll need to manually switch the kernel by going to **Runtime > Change runtime type**, and selecting R as the kernel. This allows you to run R code in the Colab environment.

However, our notebook is already configured to use R by default. Unless something goes wrong, you shouldn’t need to manually change runtime type.

### **Importing Required Packages**
**Run the following lines of code**:

In [1]:
#Do not modify

setwd("/content")

# Remove `MXB107-Notebooks` if exists,
if (dir.exists("MXB107-Notebooks")) {
  system("rm -rf MXB107-Notebooks")
}

# Fork the repository
system("git clone https://github.com/ArwenNugteren/MXB107-Notebooks")

# Change working directory to "MXB107-Notebooks"
setwd("MXB107-Notebooks")

#
invisible(source("R/preConfigurated.R"))

Loading required package: ggplot2

Loading required package: dplyr


Attaching package: ‘dplyr’


The following objects are masked from ‘package:stats’:

    filter, lag


The following objects are masked from ‘package:base’:

    intersect, setdiff, setequal, union


Loading required package: tidyr

Loading required package: stringr

Loading required package: magrittr


Attaching package: ‘magrittr’


The following object is masked from ‘package:tidyr’:

    extract


Loading required package: IRdisplay

Loading required package: png

Warning message in library(package, lib.loc = lib.loc, character.only = TRUE, logical.return = TRUE, :
“there is no package called ‘png’”
Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)

Loading required package: grid

Loading required package: knitr

Loading required package: jpeg

Warning message in library(package, lib.loc = lib.loc, character.only = TRUE, logical.return = TRUE, :
“there is no package called ‘jpeg’”
I

**Do not modify the following**

In [2]:
if (!require("testthat")) install.packages("testthat"); library("testthat")

test_that("Test if all packages have been loaded", {

  expect_true(all(c("ggplot2", "tidyr", "dplyr", "stringr", "magrittr", "knitr") %in% loadedNamespaces()))

})

Loading required package: testthat


Attaching package: ‘testthat’


The following objects are masked from ‘package:magrittr’:

    equals, is_less_than, not




Test passed with 1 success 🌈.


# 1. Discrete uniform distribution

### Example: Ice-cream shop orders
*An icecream shop serves customers one at a time. They can order one flavour out of the options: chocolate, vanilla, coffee, pistachio, strawberry and gumdrop. They also have an option to add toppings.*



### Ice-cream example part 1
The first customer is a toddler who demands to choose their own icecream flavour, but does not understand what the different flavours are. We can assume that each icecream flavour has an equal probability of being chosen.

What type of random variable is the toddler's choice of flavour? What is the probability that they choose pistachio?

<details> <summary>▶️ Click to show the solution</summary>

As each choice is equally likely, we can model this using the discrete uniform distribution. There are 6 options, so the probability of the toddler choosing pistachio is $\frac16$.

</details>

# 2. Bernoulli distribution

### Ice-cream example part 2

The toddler has an 8 year old sibling who has been promised that they can order sprinkles if they have been well-behaved. Unfortunately, they do not know if their parents are aware that they tidied up their room by stuffing everything under the bed. They estimate that they were very sneaky in doing so, and that the probability that their parents believe they have been good as 0.8.

Write down the probability distribution for getting sprinkles. What is the variance of this distribution?

<details> <summary>▶️ Click to show the solution</summary>

Let $S=1$ represent the child getting sprinkles, and $S=0$ represent them missing out. Since there are only two outcomes, $S$ can be modelled as Bernoulli random variable with parameter $p=0.8$.

For the Bernoulli distribution, we know $\text{Var}(S) = p(1-p) = 0.8 * 0.2 = 0.16$.

</details>

## Question 2.1
A lightbulb manufacturing company claims that 92% of the lightbulbs it produces pass quality inspection.
Let $X$ be a random variable defined as:

$X = 1$ if a randomly selected lightbulb passes inspection

$X=0$ if it fail to pass the inspection

a) What type of random variable is $X$?

b) Write down the parameter of the distribution of $X$.

c) Find $\mathbb{E}(X)$ and $\text{Var}(X)$.

<details> <summary>▶️ Click to show the solution</summary>

a) $X$ is a Bernoulli random variable, as there are only two possible options, passing or failing the inspection.

b) The Bernoulli distribution has parameter $p = 0.92$.

c) $$\mathbb{E}(X) = p = 0.92$$

$$\text{Var}(X) = p(1-p) = 0.92 \cdot 0.08 = 0.0736$$

</details>

## Bernoulli distribution in R
We can simulate the Bernoulli distribution in R using `rbinom(n, size=1, prob)`. This is the expression for a Binomial random variable of size 1, which is the same as a Bernoulli random variable.

In [8]:
n=10
prob = 1/6
rbinom(n,size = 1, prob)

[1] 1 0 0 0 0 0 1 0 1 0

## Question 2.2
Draw 10,000 samples from the Bernoulli distribution of lightbulbs passing quality inspection and calculate the sample mean and sample variance. Compare these values to the theoretical values calculated above.

In [9]:
N <- 10000
lightbulbs <- rbinom(N, size=1, prob=0.92)
mean(lightbulbs)
var(lightbulbs)

[1] 0.9162

[1] 0.07678524

<details>
<summary>▶️ Click to show the solution</summary>

```r
N <- 10000
prob <- 0.92

lightbulbs <- rbinom(N, size=1, prob)

mean(lightbulbs)
var(lightbulbs)

# you should find that the sample values are quite similar to the theoretical values
```
</details>

# 3. Binomial distribution

### Icecream example part 3
While you are eating your icecream (coffee with chocolate sauce on top), you are suprised to notice multiple people ordering vanilla. You estimate that the probability of ordering vanilla is 0.36.

As you continue to eat your icecream, you observe 20 more customers ordering. Let $X$ be the random variable representing whether they order vanilla.

a) What is the distribution of $X$? Write down the pmf of $X$.

b) What is the probability that exactly 4 people order vanilla? What about more than 17 people?

c) What is the expected number of vanilla orders?

In [ ]:
choose(20,18)*0.36^18*0.64^2 + choose(20,19)*0.36^19*0.64^1 + choose(20,20)*0.36^20*0.64^0

[1] 8.515754e-07

<details> <summary>▶️ Click to show the solution</summary>

a) Let $X$ be the number of customers who order vanilla. Then $X \sim \text{Bin}(20, 0.36)$. It has pmf:
$$Pr(X = k) = \binom{20}k (0.36)^k (0.64)^{20-k} \quad \text{for } k=0,\ldots,20$$

b) We need to compute $Pr(X=4)$.

$$P(X=4) = \binom{20}4 (0.36)^4 (0.64)^{16} \approx 0.0645$$

Now for the probability that more than 17 people order vanilla:
\begin{align*} P(X > 17) & = P(X = 18) + P(X=19) + P(X=20) \\
& = \binom{20}{18} 0.36^{18} 0.64^2 + \binom{20}{19} 0.36^{19} 0.64 + \binom{20}{20} 0.36^{20} \\
& \approx 8.52 \times 10^{-7}
\end{align*}

c) For the binomial distribution, $\mathbb{E}(X) = np = 20 \cdot 0.36 = 7.2$, so the expected number of vanilla orders is 7.2.
</details>

## Binomial random variables in R


In [ ]:
dbinom(4, size=20, prob=0.36)
# P(X > 17) = 1 - P(X <= 17)
1 - pbinom(10, size=20, prob=0.36)
qbinom(0.75, size=20, prob=0.36)

[1] 0.06447381

[1] 0.0645194

[1] 9


#### **R Quick Reference**

`R` supports the following functions for computing distributional quantities and simulating from binomial distributions:

- `dbinom(x, size, prob, log = FALSE)` computes PMF of `Binomial(size,p)`;
- `pbinom(q, size, prob, lower.tail = TRUE, log.p = FALSE)` computes CDF of `Binomial(size,p)`;
- `qbinom(p, size, prob, lower.tail = TRUE, log.p = FALSE)` computes `p`-quantile of `Binomial(size,p)`;
- `rbinom(n, size, prob)` simulates `n` random numbers from `n` independent r.v. following `Binomial(size,p)`.

Here,

- `x`, `q`: The number of successes.
- `size`: The number of Bernoulli trials. If `size = 1`, we obtain the Bernoulli distribution.
- `prob`: The probability of success
- `log`: Whether or not we want to return log PMF.
- `lower.tail`: If `TRUE`, return $\Pr(X \leq x)$ where $X \sim \text{Binomial(size,p)}$. If `FALSE`, return $\Pr(X > x)$.

## Question 3.1


##### **Scenario 1**

Toss a fair coin 100 times and count the number of heads. Repeat this entire experiment 10 times using simulation.

In [ ]:
coins <- rbinom(10, size=100, prob=0.5)
print(coins)
mean(coins)

 [1] 66 46 48 51 50 47 50 49 48 48


[1] 50.3

<details>
<summary>▶️ Click to show the solution</summary>
We can model the outcome of this experiment by a random variable $Y \sim \text{Binomial}(n,p)$ where $n = 100$ and $p = 0.5$. To simulate random numbers of this r.v., we can use the `rbinom()` function:

```r
rbinom(n = 10, size = 100, p = 0.5)
```
</details>

##### **Scenario 2**
A fair coin is tossed $10$ times. What is the probability of observing exactly 6 heads?

In [ ]:
dbinom(10, size=10, prob=0.5)

[1] 0.0009765625

<details>
<summary>▶️ Click to show the solution</summary>
We can model the outcome of this experiment by a random variable $Y \sim \text{Binomial}(n,p)$ where $n = 10$ and $p = 0.5$. To calculate the probability of observing exactly 6 heads, we can use the PMF of `Binomial(10,0.5)`

$$P(Y = 6) = \binom{10}{6} 0.5^{6}(1-0.5)^{10-6}.$$

We could calculate this by hand, or use `dbinom()` in R.
```r
dbinom(x = 6, size = 10, prob = 0.5)
```
</details>

##### **Scenario 3**
A fair coin is tossed $10$ times. Compute probability of observing at least 6 heads.


In [ ]:
# P(X >= 6) = 1 - P(X < 6) = 1 - P(X <= 5)
1 - pbinom(5, size=10, prob=0.5)

[1] 0.3769531

<details>
<summary>▶️ Click to show the solution</summary>
We are interested in computing:

$$\Pr(Y \geq 6).$$

This can be done using the `pbinom()` function in `R`. Note that `pbinom()` always calculates $Pr(Y \leq y)$, so we need to either calculate $1 - Pr(Y \leq 5)$ using the argument `lower.tail = TRUE` or $Pr(Y > 5)$ with the argument `lower.tail = FALSE`.  

```r
1 - pbinom(q = 5, size = 10, prob = 0.5, lower.tail = TRUE)

pbinom(q = 5, size = 10, prob = 0.5, lower.tail = FALSE)
```
</details>

##### **Scenario 4**

A factory makes 5000 light bulbs an hour (5000 per batch), and each bulb has a 1% chance of being defective. What is the number of defective bulbs that we would not exceed in 95% of the batches?



In [ ]:
qbinom(0.95, size=5000, prob=0.01)

[1] 62

<details>
<summary>▶️ Click to show the solution</summary>
We can model the number of defects in each batch by a random variable $Y \sim \text{Binomial}(n,p)$ where $n = 5000$ and $p = 0.01$.

The number of defective bulbs that we would not exceed in 95% of the batches is $95\%$-quantile of `Binomial(n,p)`, which can be computed using the `qbinom()` function.

```r
qbinom(p = 0.95, size = 5000, prob = 0.01, lower.tail = TRUE)
```
</details>

# 4. Geometric distribution

### Ice-cream example part 4
Customers order toppings independently with probability 0.4. Let $Y$ be the number of customers up till and including the first customer of the day who orders a topping.

a) Compute the probability that $Y=4$ and $Y \geq 3$.

b) Calculate $\mathbb{E}(Y)$ and interpret this value.

<details> <summary>▶️ Click to show the solution</summary>

We can model $Y$ as a geometric random variable with parameter $p = 0.4$, giving us the pmf $P(Y = k) = 0.6^{k-1} 0.4$ for $k \geq 1$.

a) $$Pr(Y = 4)  = 0.6^3 \cdot 0.4 = 0.0864$$

\begin{align*} Pr(Y \geq 3) & = 1 - Pr(Y \leq 2) = 1 - (Pr(Y=2) + Pr(Y=1)) \\
& = 1 - (0.6^1 \cdot 0.4 + 0.6^0 \cdot 0.4) \\
& = 1 - 0.64 = 0.36 \end{align*}

b) $$\mathbb{E}(Y) = \frac1p = \frac1{0.4} = 2.5$$
On average, we expect the first topping order to be from the second or third customer of each day.


</details>

## Geometric distribution in R

#### **R Quick Reference**

`R` supports the following functions for computing distributional quantities and simulating from geometric distributions:

- `dgeom(x, prob, log)` computes PMF of `Geometric(prob)`;  
- `pgeom(q, prob, lower.tail = TRUE)` computes CDF of `Geometric(prob)`;  
- `qgeom(p, prob)` computes `p`-quantile of `Geometric(prob)`;  
- `rgeom(n, prob)` simulates `n` random numbers from `Geometric(prob)

Importantly, the built-in geometric distribution in R counts the *number of failures* before success. If you need to calculate the total number of trials, you'll need to add 1 to `x`.

## Question 4.1: does Brisbane have good coffee?
You have decided to embark on a quest to find the best coffee in Brisbane near uni. While pessimistic about the possibility of finding a truly good coffee shop, you decide that a coffee rated 4.5/5 is acceptably good to end your search. Most coffee shops are pretty average, but a wise friend has indicated that they consider that about 1 in 21 coffee shops meet your standards.

You decide to visit a different shop every morning until you finally find a good one.

a) What is the probability that you visit exactly 6 terrible shops before finding a good one on your 7th day?

b) You realise there are only 15 coffee shops within 10 minutes walk of campus. What is the probability that you find a 4.5/5 coffee spot before needing to extend your search further afield?

c) Use `rgeom()` to simulate 1,000 students' quests. Suppose that every "bad" coffee costs you $5.50. What is the average amount of money wasted on bad coffee before finding a good spot?

<details>
<summary>▶️ Click to show the solution</summary>

```r
# Probability of finding a good shop
p_coffee <- 1/21

# a) Exactly 6 failures (bad shops) before the success
# remember that dgeom calculates number of failures
prob_6_bad <- dgeom(x = 6, prob = p_coffee)
print(prob_6_bad)

# b) Success within 15 shops total
# This means failures (q) can be anything from 0 to 14
prob_success_14 <- pgeom(q = 14, prob = p_coffee)
print(prob_success_14)

# c) Financial Simulation
# Simulate number of failures for 1,000 students
failures <- rgeom(n = 1000, prob = p_coffee)

# Calculate wasted money (failures * $5.50)
wasted_money <- failures * 5.50
avg_waste <- mean(wasted_money)

print(avg_waste)
# is this amount worth it?
```

</details>

# 5. Negative binomial distribution

### Icecream example part 5
You are working at the icecream shop and notice that you are running low on chocolate flavour. You only have enough to make 2 more orders. If you run out, you will close the store to avoid customers becoming angry over the lack of chocolate. You estimate that the probability of a customer ordering chocolate is 0.27.

a) How many more customers do you expect to serve before you run out and close the store?

b) The store is only open for another 30 minutes anyway, during which period you estimate you can serve one customer per minute. What is the probability that you close the store at closing time without running out of chocolate first?

<details> <summary>▶️ Click to show the solution</summary>

Let $X$ be the number of customers until the second chocolate order. $X$ follows a negative binomial distribution with $k=2$ and $p=0.27$.

a) $$\mathbb{E}(X) = \frac{k}{p} = \frac{2}{0.27} \approx 7.41$$
You expect to serve 7 or 8 customers before running out of chocolate.

b) This time we can model the number of customers ordering chocolate as $Y$, a regular binomial distribution with 30 total customers and probability 0.27 of ordering chocolate. Then we want to find the probability that less than 2 out of 30 customers order chocolate.
\begin{align*}
Pr(Y < 2) & = P(Y=0) + P(Y=1) \\
& = (1-0.27)^30 + 30 \cdot 0.27 \cdot (1-0.27)^{29} \\
& \approx 0.00096
\end{align*}
So there is only a 0.096% chance that you make it to closing time without needing to close early because you ran out of chocolate.

</details>

## Question 5.1
The probability that a randomly selected box of a certain type of cereal has a particular prize is 0.2. Suppose you purchase box after box until you have obtained two of these prizes.


a) What is the probability that you purchase $x$ boxes until you obtain the two prizes?

b) What is the probability that you only have to purchase four boxes?

c) What is the probability that you have to purchase more than four boxes?

d) How many boxes without the desired prize do you expect to purchase?

<details> <summary>▶️ Click to show the solution</summary>

Let $X$ be the number of boxes we must purchase until we obtain two prizes. Then $X \sim \mathrm{NegBin}(2,0.2)$.


a) $$Pr(X = x) = \binom{x-1}{1} 0.2^2 0.8^{x - 2},\quad x \ge 2$$


b) \begin{align*}
 				Pr(X = 4) = \binom{4-1}{1} 0.2^2 0.8^{4-2} = 0.0768.
 			\end{align*}

c) \begin{align*}
				Pr(X > 4) &= 1 - Pr(X \le 4).\\
 				Pr(X \le 4) &= Pr(X = 2) + Pr(X = 3) + Pr(X = 4),\\
 					&= \binom{2-1}{1} 0.2^{2} 0.8^{2-2} + \binom{3-1}{1} 0.2^{2} 0.8^{3-2} + \binom{4-1}{1} 0.2^{2} 0.8^{4-2},\\
 					&= \binom{1}{1} 0.2^2 0.8^0 + \binom{2}{1} 0.2^2 0.8^1 + \binom{3}{1} 0.2^2 0.8^2,\\
 					&= 0.1808.\\
 				\therefore Pr(X > 4) &= 1 - 0.1808,\\
 					&= 0.8192
 			\end{align*}

d) We are interested in $\mathbb{E}(X - 2)$:
			\begin{align*}
 				\mathbb{E}(X-2) &= \mathbb{E}(X) - 2,\\
 					&= \dfrac{k}{p}  - 2 \quad \text{with } k=2,\:p=0.2,\\
 					&= \dfrac{2 }{0.2} - 2,\\
 					&= 8.
		 	\end{align*}
Alternatively, we could have parametrised differently, and let $Y$ be the *number of failures* until 2 successes. In this case, we would have used the other form of the negative binomial distribution for $Pr(Y = y)$, and in (d) we would look for $\mathbb{E}(Y)$.

</details>

# 6. Poisson distribution

### Ice-cream example part 6
Let $N$ be the number of toppings ordered in an hour.

a) Given an average rate of one customer per minute, and probability of ordering toppings of 0.4, calculate the average rate of toppings per hour. Compute $P(N = 15)$.  

b) We now consider a 15-minute interval. What is the corresponding rate parameter for toppings? What is the probability of no topping orders in this interval?



<details> <summary>▶️ Click to show the solution</summary>

We can model $N$ using a Poisson distribution.

a) Over the hour we expect $\lambda = 60 \times 0.4 = 24$ toppings/hour, and this becomes the parameter of our Poisson distribution, i.e. $N \sim \text{Poisson}(24)$.

$$Pr(N=15) = \frac{24^{15} e^{-24}}{15!} \approx 0.015$$

b) In a 15-minute interval, we expect $24 \times 0.25 = 6$ toppings/15 minutes (as 15 minutes is a quarter of an hour).  Let $X$ be the number of topping orders in 15 minutes, $X \sim \text{Poisson}(6)$

$$Pr(X = 0) = \frac{6^0 e^{-6}}{0!} \approx 0.0025$$

</details>

## Poisson Distribution in R






#### **R Quick Reference**

`R` supports the following functions for computing distributional quantities and simulating from Poisson distributions:

- `dpois(x, lambda, log = FALSE)` computes PMF of `Poisson(lambda)`;  
- `ppois(q, lambda, lower.tail = TRUE, log.p = FALSE)` computes CDF of `Poisson(lambda)`;  
- `qpois(p, lambda, lower.tail = TRUE, log.p = FALSE)` computes `p`-quantile of `Poisson(lambda)`;  
- `rpois(n, lambda)` simulates `n` random numbers from `Poisson(lambda)`.

Here:

- `x`, `q`: Number of events.  
- `lambda`: Expected number of events.  
- `log`: Whether or not to return log PMF.  
- `lower.tail`: If `TRUE`, return $\Pr(Y \leq x)$; if `FALSE`, return $\Pr(Y > x)$.

## Question 6.1

The number of emails a server receives in an hour is modeled as a Poisson random variable with $\lambda = 5$.



##### **Exercise 1**

Simulate the number of emails received in 10 separate hours.


<details>
<summary>▶️ Click to show the solution</summary>

```r
rpois(n = 10, lambda = 5)
```

</details>

##### **Exercise 2**

What is the probability that exactly 3 emails are received in one hour?

<details>
<summary>▶️ Click to show the solution</summary>
These are equivalent solutions

```r
(5^3 * exp(-5)) / factorial(3)
```

```r
dpois(x = 3, lambda = 5)
```

</details>

##### **Exercise 3**
Calculate the probability of receiving between 3 and 6 emails inclusive in an hour.

<details>
<summary>▶️ Click to show the solution</summary>
These are equivalent solutions

```r
ppois(6, lambda = 5) - ppois(2, lambda = 5)
```

```r
sum((5^(3:6) * exp(-5)) / factorial(3:6))
```

</details>

# Workshop questions
Your turn to practice! Work on these individually or discuss with others near you, or for extra practice at home.

## Question 1

A factory produces 10,000 light bulbs per hour, with each bulb having a 0.05% chance of being defective. Defects occur independently, and each batch of bulbs (produced each hour) is independent of other batches.

A quality control team checks each batch hourly. A batch passes the test if it contains at most 10 defective bulbs; otherwise, the batch fails.
Consider a year with 365 days, and assume 24 production hours per day.




#### **Question 1.1**

Let $Y$ be the number of batches that fail in one year. Determine the distribution of $Y$ and its mean and variance.



#### **Question 1.2**
Calculate the exact probability at most 125 batches fail in the year.

#### **Question 1.3**
Generate $n = 10^6$ samples from this distribution. Describe the shape of the simulated distribution.

<details>
<summary>▶️ Click to show the solution</summary>

Solutions will be released at the end of the week!

</details>

## Question 2

A company that produces fine crystal knows from experience that 10\% of its goblets have cosmetic flaws and must be classified as "seconds". Among six randomly selected goblets:


a) What is the probability that only one is a second?

b) What is the probability that at least two are seconds?

c) Calculate the expected number to be seconds and the variance for the number of seconds.

<details>
<summary>▶️ Click to show the solution</summary>

Solutions will be released at the end of the week!

</details>

## Question 3
The article "Expectation analysis of the probability of failure for water supply pipes" (J Pipeline Syst Eng, May 2012:36-46) proposed using the Poisson distribution to model the number of failures in pipelines of various types. Suppose that for cast-iron pipe of a particular length, the expected number of failures is 1 (very close to one of the cases considered in the article). Then $X$, the number of failures, has a Poisson distribution with $\lambda = 1$.


a) Find $Pr(X \le 5)$

b) Find $Pr(X = 2)$

c) Find $Pr(2 \le X \le 4)$

d) What is the probability that $X$ exceeds its mean value by more than one standard deviation?

<details>
<summary>▶️ Click to show the solution</summary>

Solutions will be released at the end of the week!

</details>

## Question 4
A vending machine has six different snack options, each labelled with a button number from 1 to 9. A student, feeling indecisive, decides to close their eyes and press one button at random, giving each button an equal chance of being selected.

### Question 4.1
Choose a distribution to model this random variable (the button that the student presses). Find the expected value of this random variable.



<details>
<summary>▶️ Click to show the solution</summary>

Solutions will be released at the end of the week!

</details>

### Question 4.2
The student's friend claims that the expected value is the button that they are more likely to press than any other. Why is this statement incorrect?

<details>
<summary>▶️ Click to show the solution</summary>

Solutions will be released at the end of the week!

</details>

## Question 5
Given the probability of successfully winning the jackpot in Oz lotto is 1/45379620, write an expression for the probability of winning the Oz lotto jackpot within a single year (52 weeks) if you purchase a ticket once every week using:


a) The geometric distribution.

b) The binomial distribution.

In both cases, assume that the draw each week is independent of the other draws. Feel free to use R as a calculator to evaluate your expression.

<details>
<summary>▶️ Click to show the solution</summary>

Solutions will be released at the end of the week!

</details>

## Question 6

A factory makes 10000 light bulbs an hour (10000 per batch), and each bulb has a 5% chance of being defective.


- What is the number of defective bulbs that we would not exceed in 95% of the batches?
- A quality control expert wants to answer the question. However, there is no built-in `qbinom()` function on their computer, so they use a simulation-based approach to answer the question:
  - Simulate $10^6$ random numbers from the corresponding Binomial distribution.
  - Estimate the 95%-quantile using the sample data.  
  - Compute the sample mean and sample variance and compare them to the theoretical values.








<details>
<summary>▶️ Click to show the solution</summary>

Solutions will be released at the end of the week!

</details>

## Question 7
An airline company accepts 105 bookings for 100-seater planes, knowing that each passenger has a 10\% chance of failing to turn up for a flight.  Find an expression for the probability that the airline company will be unable to seat all the passengers who arrive for the flight having booked a seat.

<details>
<summary>▶️ Click to show the solution</summary>

Solutions will be released at the end of the week!

</details>

## Question 8
The probability of the internet dropping out on a specific day is 0.2. Calculate the following probabilities.


a) Not having the internet drop out on a single day.

b) No interruptions to the internet for (at least) the next two days.

<details>
<summary>▶️ Click to show the solution</summary>

Solutions will be released at the end of the week!

</details>

## Question 9
The number of machine failures per day in a certain plant has a Poisson distribution with a mean of 2. Present facilities at the plant can repair two machines per day. Failures in excess of two are repaired by an outside contractor.


a) What is the probability, on a given day, of having machines repaired by the contractor?

b) What is the probability that the contractor is required at least one day in a 5-day working week? You may assume that the number of failures is independent between days.

<details>
<summary>▶️ Click to show the solution</summary>

Solutions will be released at the end of the week!

</details>

## R modelling activity
Simulate different discrete distributions in R and plot histograms of their shape. How does changing their parameters change the distribution?